lib/TokenManager.js

```diff
@@ -141,29 +139,27 @@ function renew(sdk, tokenMgmtRef, storage, key) {
   if (!tokenMgmtRef.renewPromise[key]) {
     tokenMgmtRef.renewPromise[key] = sdk.token.renew(token)
     .then(function(freshTokens) {
-      // We may receive more tokens than we requested
-      var map = {};
-      if (freshTokens instanceof Array === false) {
-        freshTokens = [freshTokens];
+      var freshToken = freshTokens;
+      // With PKCE flow we will receive multiple tokens. Find the one we are looking for
+      if (freshTokens instanceof Array) {
+        freshToken = freshTokens.filter(function(freshToken) {
```

Comment:
`freshTokens.find()` would avoid the hidden(ish) `pop()` at the end.

Reply:
yes good suggestion